### PR TITLE
Standardize AWS ECS naming

### PR DESCRIPTION
**Pull request by ferruzzi**
Part of https://github.com/apache/airflow/issues/20296

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/amazon/aws/operators/ecs.py [hunk 1]
```diff
@@ -508,3 +509,50 @@ def on_kill(self) -> None:
...
+        warnings.warn(
+            "This class is deprecated. "
+            "Please use `airflow.providers.amazon.aws.operators.ecs.EcsTaskLogFetcher`.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        super().__init__(*args, **kwargs)
+
+
+class ECSProtocol(EcsProtocol, Protocol):
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/operators/ecs.py:546]: @eladkal @uranusjr 

I'm not positive on this one.   It does pass pre-commits and a local CI run like this, but let me know if this should have been handled differently.

Two things happened here:  
1. Without inheriting 'Protocol' I got an error that only Protocols could inherit another Protocol, so I added that inheritence here.

2. The EcsProtocol doesn't have an __init__ so I'm not sure this is actually correct to call super().__init__().

File: airflow/providers/amazon/aws/operators/ecs.py [hunk 2]
```diff
@@ -508,3 +509,50 @@ def on_kill(self) -> None:
...
+    Please use :class:`airflow.providers.amazon.aws.operators.ecs.EcsProtocol`.
+    """
+
+    def __init__(self):
+        warnings.warn(
+            "This class is deprecated. "
+            "Please use `airflow.providers.amazon.aws.operators.ecs.EcsProtocol`.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/operators/ecs.py]: @eladkal @uranusjr

I'm not positive on this one. It does pass pre-commits and a local CI run like this, but let me know if this should have been handled differently.

Without inheriting 'Protocol' I got an error that only Protocols could inherit another Protocol, so I added that inheritence here.  Also, ECSProtocol didn't have an init method so I added one just for the deprecation message, but I'm not sure if that as the right call.

**Comment by ferruzzi** [comment]
Rebased to resolve merge conflicts.

File: airflow/providers/amazon/aws/operators/ecs.py (hunk 2 above)
Comment (ferruzzi) [reply airflow/providers/amazon/aws/operators/ecs.py]: See Also:  https://github.com/apache/airflow/pull/20369/files#r771158731

**Comment by eladkal** [comment]
We also have
https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/exceptions.py

File: airflow/providers/amazon/aws/exceptions.py [hunk 3]
```diff
@@ -18,12 +18,29 @@
...
 
 
-class ECSOperatorError(Exception):
+class EcsOperatorError(Exception):
     """Raise when ECS cannot handle the request."""
 
     def __init__(self, failures: list, message: str):
         self.failures = failures
         self.message = message
         super().__init__(message)
```
Comment (ferruzzi) [inline airflow/providers/amazon/aws/exceptions.py:30]: Are you looking for this @eladkal ?

Comment (eladkal) [reply airflow/providers/amazon/aws/exceptions.py:30]: ahh missed it

File: airflow/providers/amazon/aws/operators/ecs.py (hunk 2 above)
Comment (eladkal) [reply airflow/providers/amazon/aws/operators/ecs.py]: I checkout your branch and i'm unable to reproduce the behavior that you are describing.

**Comment by ferruzzi** [comment]
I was away for the holidays, but I think this one is ready to go?

**Comment by eladkal** [comment]
> I was away for the holidays, but I think this one is ready to go?

see https://github.com/apache/airflow/pull/20332#discussion_r772515498
I wasn't able to reproduce what you described

**Comment by ferruzzi** [comment]
Huh.   Neither can I now.    I've rebased and removed the `Protocol` inheritance on L546.

**Review by eladkal — APPROVED** [review]
LGTM
will merge when CI is green

**Pull request merged**